src/OrchardCore.Modules/OrchardCore.Localization/Startup.cs

```diff
@@ -73,4 +77,41 @@ public override void ConfigureServices(IServiceCollection services)
             services.Configure<RequestLocalizationOptions>(options => options.ApplyCurrentCultureToResponseHeaders = true);
         }
     }
+
+    [Feature("OrchardCore.Localization.CulturePicker.Admin")]
+    public class CulturePickerStartup : StartupBase
+    {
+        private static readonly Task<ProviderCultureResult> NullProviderCultureResult = Task.FromResult(default(ProviderCultureResult));
+
+        private readonly AdminOptions _adminOptions;
+
+        public CulturePickerStartup(IOptions<AdminOptions> adminOptions)
+        {
+            _adminOptions = adminOptions.Value;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<IShapeTableProvider, AdminCulturePickerShapes>();
+
+            services.Configure<RequestLocalizationOptions>(options => options.AddInitialRequestCultureProvider(new CustomRequestCultureProvider(context =>
+            {
+                var path = context.Request.Path.Value[1..];
+                if (path.Equals(_adminOptions.AdminUrlPrefix, StringComparison.OrdinalIgnoreCase) ||
+                    path.StartsWith(_adminOptions.AdminUrlPrefix + "/", StringComparison.OrdinalIgnoreCase))
+                {
+                    var cookie = context.Request.Cookies[LocalizationCookieName.AdminSite];
```

Comment:
This only seems to stick if you visit pages under Settings:

https://user-images.githubusercontent.com/1976647/181098265-f44321df-77fb-4e54-9b74-6339e43efce3.mp4

Reply:
Seems the URL check after address the feedback has some limitation. Let me check

Reply:
Seems the cookie path is matter here in javascript, basically is attach the cookie to the page by default not the domain. I just fix that by setting the path to tenant path

https://user-images.githubusercontent.com/3237266/181163271-5a730432-9bed-4e75-a2ed-7374e96fa06b.mp4